File: ckgame-java/src/main/java/com/ckgame/events/EventInstance.java
package com.ckgame.events;

import java.util.List;

/**
 * 待处理事件实例。
 */
public final class EventInstance {
    public final int eventId;
    public final int character;
    public final String title;
    public final String description;
    public final List<EventChoice> choices;

    public EventInstance(int eventId, int character, String title,
                         String description, List<EventChoice> choices) {
        this.eventId = eventId;
        this.character = character;
        this.title = title;
        this.description = description;
        this.choices = choices;
    }
}

File: ckgame-ts/src/events/EventInstance.ts
import { EventChoice } from './EventChoice.js';

/**
 * A pending event instance for a specific character.
 */
export class EventInstance {
  public readonly eventId: number;
  public readonly character: number;
  public readonly title: string;
  public readonly description: string;
  public readonly choices: EventChoice[];

  constructor(
    eventId: number,
    character: number,
    title: string,
    description: string,
    choices: EventChoice[],
  ) {
    this.eventId = eventId;
    this.character = character;
    this.title = title;
    this.description = description;
    this.choices = choices;
  }
}
